v4/export/writer_util.go

```diff
@@ -12,6 +14,62 @@ import (
 	"go.uber.org/zap"
 )
 
+const lengthLimit = 1048576
+
+type Dao struct {
+	bp sync.Pool
+}
+
+func NewDao() (d *Dao) {
+	d = &Dao{
+		bp: sync.Pool{
+			New: func() interface{} {
+				return &bytes.Buffer{}
+			},
+		},
+	}
+	return
+}
+
+type writerPipe struct {
+	sync.Mutex
+
+	input  chan string
+	closed chan struct{}
+
+	w   io.StringWriter
+	err error
+}
+
+func (b *writerPipe) Run(ctx context.Context) {
+	defer close(b.closed)
+	for {
+		select {
+		case s, ok := <-b.input:
+			if !ok {
+				return
+			}
+			if b.err != nil {
+				continue
+			}
+			err := write(b.w, s)
```

Comment:
same problem, sometimes with `io.Stringer` and sometime with `bytes.Buffer`

Reply:
`io.Stringer` only write string here.
`bytes.Buffer` sometimes write bytes sometimes writes string